r/src/compute.cpp

```diff
@@ -618,6 +622,9 @@ SEXP compute__CallFunction(std::string func_name, cpp11::list args, cpp11::list
 
 // [[arrow::export]]
 std::vector<std::string> compute__GetFunctionNames() {
+#if ARROW_VERSION_MAJOR >= 21
+  auto compute_init_status_ = arrow::compute::Initialize();
```

Comment:
How about checking that no error occurred?

Reply:
Also in c_glib